back/app/services/permissions_service.rb

```diff
@@ -123,6 +123,7 @@ def denied_reason(permission, user)
         return if UserRoleService.new.can_moderate? permission.permission_scope, user
         return DENIED_REASONS[:not_permitted] if permission.permitted_by == 'admins_moderators'
 
+        return DENIED_REASONS[:missing_data] if any_custom_field_required?(requirements(permission, user))
```

Comment:
@sebastienhoorens Checking if the current approach makes sense before writing tests.

Context: survey that's only allowed for users in a certain smart group based on a registration question value. I want to participate, sign up and close the step with custom fields:

<img width="715" alt="image" src="https://github.com/CitizenLabDotCo/citizenlab/assets/16427929/c02fe0c4-5476-4cf5-bd5f-e5bafda630fc">

The button is now disabled and I have to figure out myself I need to complete a registration field by going to the user menu > complete your profile:

<img width="404" alt="image" src="https://github.com/CitizenLabDotCo/citizenlab/assets/16427929/402e2d2d-0e31-4625-ae1d-018f0e2ca6a7">

===

The suggested changes check for missing data *before* the 'is in group' part, so the button to Take the survey is not disabled and the Complete your profile modal is triggered again, to fill out the required custom field.

So TL;DR: we want to push a user to fill out all (required) fields, because they can become part of a smart group and gain right to participate.